test/Core.Test/OrganizationFeatures/OrganizationCollections/BulkAddCollectionAccessCommandTests.cs

```diff
@@ -0,0 +1,246 @@
+            .Returns(users);
+
+        sutProvider.GetDependency<IGroupRepository>()
+            .GetManyByManyIds(Arg.Any<IEnumerable<Guid>>())
+            .Returns(groups);
+
+        var exception = await Assert.ThrowsAsync<BadRequestException>(() => sutProvider.Sut.AddAccessAsync(org.Id, collections,
+            ToAccessSelection(collectionUsers),
+            ToAccessSelection(collectionGroups)
+        ));
+
+        Assert.Contains("One or more groups do not exist.", exception.Message);
+
+        await sutProvider.GetDependency<IOrganizationUserRepository>().ReceivedWithAnyArgs().GetManyAsync(default);
+        await sutProvider.GetDependency<IGroupRepository>().ReceivedWithAnyArgs().GetManyByManyIds(default);
+    }
+
+    [Theory, BitAutoData, CollectionCustomization]
+    public async Task ValidateRequestAsync_GroupWrongOrg_Failure(SutProvider<BulkAddCollectionAccessCommand> sutProvider,
+        Organization org,
+        IList<Collection> collections,
+        IList<OrganizationUser> users,
+        IList<Group> groups,
+        IEnumerable<CollectionUser> collectionUsers,
+        IEnumerable<CollectionGroup> collectionGroups)
+    {
+        groups.First().OrganizationId = Guid.NewGuid();
+
+        sutProvider.GetDependency<IOrganizationUserRepository>()
+            .GetManyAsync(Arg.Any<IEnumerable<Guid>>())
+            .Returns(users);
+
+        sutProvider.GetDependency<IGroupRepository>()
+            .GetManyByManyIds(Arg.Any<IEnumerable<Guid>>())
+            .Returns(groups);
+
+        var exception = await Assert.ThrowsAsync<BadRequestException>(() => sutProvider.Sut.AddAccessAsync(org.Id, collections,
+            ToAccessSelection(collectionUsers),
+            ToAccessSelection(collectionGroups)
+        ));
+
+        Assert.Contains("One or more groups do not belong to the same organization as the collection being assigned.", exception.Message);
+
+        await sutProvider.GetDependency<IOrganizationUserRepository>().ReceivedWithAnyArgs().GetManyAsync(default);
+        await sutProvider.GetDependency<IGroupRepository>().ReceivedWithAnyArgs().GetManyByManyIds(default);
+    }
+
+    private static ICollection<CollectionAccessSelection> ToAccessSelection(IEnumerable<CollectionUser> collectionUsers)
+    {
+        return collectionUsers.Select(cu => new CollectionAccessSelection
+        {
+            Id = cu.OrganizationUserId,
+            Manage = cu.Manage,
+            HidePasswords = cu.HidePasswords,
+            ReadOnly = cu.ReadOnly
+        }).ToList();
+    }
+    private static ICollection<CollectionAccessSelection> ToAccessSelection(IEnumerable<CollectionGroup> collectionGroups)
+    {
+        return collectionGroups.Select(cg => new CollectionAccessSelection
+        {
+            Id = cg.GroupId,
+            Manage = cg.Manage,
+            HidePasswords = cg.HidePasswords,
+            ReadOnly = cg.ReadOnly
+        }).ToList();
+    }
```

Comment:
It looks like some `CollectionRepository` methods do a transformation like this (e.g. see `GetManyByOrganizationIdWithAccessAsync`), maybe we could move this into production code? e.g. as a method on `CollectionUser`/`Group` or as a constructor overload on `CollectionAccessSelection`.

Reply:
I like this idea; however, as I started doing so, I re-discovered the `CollectionAccessSelection.Id` property flips what it is identifying based on the perspective of the "owning" entity of the list of `CollectionAccessSelection`.

In the above code, the perspective is from a single collection that owns multiple `CollectionAccessSelection` for various Group/Users, so the `Id` property refers to a `OrganizationUserId` or `GroupId`.

However, this relationship flips when a Group or OrgUser owns multiple `CollectionAccessSelection` and the `Id` property now refers to a `ColllectionId`. See below for an example in the `GroupRepository.cs`.  https://github.com/bitwarden/server/blob/2aaef3cf64119440be897e937c0f68a40f27328a/src/Infrastructure.Dapper/Repositories/GroupRepository.cs#L69-L75

So, because we use the same entity for both perspectives, it makes it tricky to create a method or constructor overload. We would need to take another argument or create additional overloads to specify if the `Id` property should be a `CollectionId` OR a `GroupId`/`OrganizationUserId`. I don't really like either of those options at this point, and think this further highlights that we should look into renaming/refactoring these `CollectionAccessSelection` objects at some point (seems out of scope for this ticket, but is probably worthwhile for Flexible Collections).

Let me know what you think!

Reply:
That's messy that the `Id` can refer to different things. Let's leave the explicit initialization, it seems safest.